Main/OOP18/drink_factory.cpp

```diff
@@ -0,0 +1,16 @@
+#include "drink_factory.h"
+
+Drink* createDrink(DrinkType type, SizeType size) {
+    switch (type) {
+    case DrinkType::ESPRESSO:
+    case DrinkType::AMERICANO:
+    case DrinkType::LATTE:
+    case DrinkType::CAPPUCCINO:
+        return new Coffee(type, size);
```

Comment:
convert here DrinkType to CoffeeType and pass CoffeeType as a c-tor parameter instead